### PR TITLE
feat(make-pain-visible): backend

### PR DESCRIPTION
**Pull request by macobo**
## Problem

We want to expose who and when and what are having frustrating experience (performance-wise) using the app.

## Changes

This PR creates API endpoints for getting information on sessions, interactions and queries and does some scaffolding to tie it into data exploration.

Later prs (featuring @pauldambra) will follow this up and create a proper frontend.

Note that using this requires setting up the schema manually: Run queries you get from `python manage.py shell <  posthog/models/query_metrics/sql.py`

## How did you test this code?

Open localhost:8000/query, play with it

### CODE REVIEW

File: frontend/src/queries/nodes/TimeToSeeData/timeToSeeDataLogic.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    path(['queries', 'nodes', 'TimeToSeeData', 'timeToSeeDataLogic']),
+    props({} as TimeToSeeDataLogicProps),
+    key((props) => props.key),
+    loaders(({ props }) => ({
+        response: [
+            null as TimeToSeeNode | null,
+            {
+                loadData: async () => {
+                    const response = await query(props.query)
+                    // TODO: Resolve this typing mess
```
Comment (pauldambra) [inline frontend/src/queries/nodes/TimeToSeeData/timeToSeeDataLogic.ts:24]: 🤣

File: frontend/src/queries/schema.ts [hunk 2]
```diff
@@ -269,6 +277,30 @@ export interface UnimplementedQuery extends InsightsQueryBase {
...
+    teamId?: number
+}
+
+export interface TimeToSeeDataQuery extends DataNode {
+    kind: NodeKind.TimeToSeeDataQuery
+
+    /** Project to filter on. Defaults to current project */
+    teamId?: number
+
+    /** Project to filter on. Defaults to current session */
```
Comment (pauldambra) [inline frontend/src/queries/schema.ts:296]: Not the project I think... but we can clean that in follow-up

**Review by pauldambra — APPROVED** [review]
LGTM, nothing I can see here on the query/visualisation side that is painting us into a corner

File: frontend/src/queries/nodes/TimeToSeeData/timeToSeeDataLogic.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                    return (response ?? null) as any as TimeToSeeNode | null
+                },
+            },
+        ],
+    })),
```
[the remark this answers is not in the document]
Comment (macobo) [reply frontend/src/queries/nodes/TimeToSeeData/timeToSeeDataLogic.ts:29]: :shrug: Will leave it to Paul in the follow-up

**Pull request merged**